### ISSUE TITLE
Deploy only when create a new release/tag

### ISSUE THREAD

**Issue by fernandoneto** [Strider-CD/strider#736]
i would like to know if is possible configure strider to deploy using  github tags instead of using directly branch's.

**Comment by knownasilya** [comment]
@fernandoneto sounds like a nice idea, although not currently supported. You might be able to write a bash script an use it in the custom scripts plugin..

**Comment by fernandoneto** [comment]
@knownasilya thank you for your fast answer.
about the bash script idea i cant get your train of thought. can you please be more verbose? :)

**Comment by knownasilya** [comment]
You could use something like [this](http://stackoverflow.com/a/1593574/483616) in a bash if statement in the [strider-custom](https://github.com/Strider-CD/strider-custom)'s (installed by default, enable it in the branches plugins) deploy section.

**Comment by fernandoneto** [comment]
@knownasilya thank you for your help :+1:

**Comment by fernandoneto** [comment]
@knownasilya in fact i think i express myself wrong. what i really want is after create a new tag based in a branch, strider is able to deploy it.

so my question is : If i create a tag in Github, strider will be notify and run based in that tag?

let me give you an example:

You and your team work in a branch for some time, every time you push commits strider run tests. So you want to send the code to productin servers. You create a tag based in that branch, and strider will deploy the code. maybe this can be possible using webwooks i think ???

hope you understand my idea :)

**Comment by knownasilya** [comment]
Ah, so you're wondering if pushing a tag will kick off a job in strider? Not sure..

**Comment by fernandoneto** [comment]
Yep thats it. 
maybe a plugin that give this options would be very usefull don't you think?

**Comment by knownasilya** [comment]
Sounds like a feature for [strider-git](https://github.com/Strider-CD/strider-git).

**Comment by knownasilya** [comment]
https://github.com/Strider-CD/strider-github/blob/master/lib/api.js#L42

**Comment by bitwit** [comment]
I can confirm new tags kick off a job in Strider, at least with Github and Bitbucket. I'm not sure if it is at the provider's discretion whether or not to push that info in a webhook.

I believe only Github starts jobs for Pull Requests, but I don't think so for Bitbucket

**Comment by knownasilya** [comment]
@bitwit just tags by themselves, i.e. `git push --tags`?

**Comment by bitwit** [comment]
Yep, I just tested it a couple times even. They don't come back with anything in the 'branch' column though. Not sure about what the Strider metadata contains either. 

![screen shot 2015-02-20 at 3 54 16 pm](https://cloud.githubusercontent.com/assets/707507/6309208/decfeda6-b918-11e4-9145-66eb8cc91391.png)

**Comment by knownasilya** [comment]
Interesting, I think it's registered as a commit, because that's the default.

So yeah, builds no longer make sense to be focused on a branch so tightly.

**Comment by fernandoneto** [comment]
@knownasilya i already have strider start from release new tags. my proble now is the same as @bitwit . i can get any information in branch column. 
![captura de ecra de 2015-03-12 15 23 52](https://cloud.githubusercontent.com/assets/6515784/6621026/0d9572ec-c8cc-11e4-85f2-c94090f5000c.png)

Could you tell me the file i found that confguration?

**Comment by knownasilya** [comment]
@fernandoneto tags are not placed on branches, so you can't have branch info. This is why fundamentally strider needs to be reworked to have builds not heavily tied to branches.

**Comment by knownasilya** [comment]
Although, github releases might be tied to branches.

**Comment by cusspvz** [comment]
Hey @knownasilya !

@fernandoneto is my co-worker/mate!! :)
I asked him to place and follow up this discussion so we can fire releases to our production servers directly from **GitHub** releases instead of branch updates. Thanks for bringing up discussion about this so far!

> github releases might be tied to branches

**GitHub** releases are created on top of git tags, which means that you couldn't have a **GitHub** release without having a git tag.

Since **Strider** is catching up git tags as branches, could the configuration be as simple as creating settings for `refs/tags/*` ?

![captura de ecra 2015-03-12 as 23 13 53](https://cloud.githubusercontent.com/assets/3604053/6630152/95d63e02-c90d-11e4-8b71-10ff6757818b.png)

**Comment by cusspvz** [comment]
Note: I didn't tested it, but i think it might work.

**Comment by knownasilya** [comment]
So that most likely will work, but the problem is that strider expects a certain output to keep the reference of your build, and this output happens to be the branch information. For tags to be supported, properly, strider needs some changes. Currently it will probably hit the `*` "branch" when you use a tag.

**Comment by cusspvz** [comment]
> but the problem is that strider expects a certain output to keep the reference of your build, and this output happens to be the branch information.

What kind of branch information?

Just tested a `git checkout` directly from `refs/tags/...` and git sees it as `detached HEAD`. If **Strider** is okay with that, we just need to access to some kind of build history to avoid **Strider** to build already built tags (in case we mess up with something).

**Comment by knownasilya** [comment]
Basically it strider needs something to associate the tag to a build, which is what is done with branches, it's associated by the branch name. Since tags are different every time, we'd need a tag pattern, like a regex or something.

Something like `{semver}-production`, where `v1.2.3-production` would trigger that build.

**Comment by cusspvz** [comment]
Well, I spent a couple of minutes trying to make a semver-compatible regexp (already with version descriptors checking) in case you need it.

`^[0-9]{1,3}.[0-9]{1,3}.[0-9]{1,3}(?:-[a-z0-9\-]*)?$`
[some acceptance testing](https://regex101.com/r/dR6oD4/1)

For my understanding perspective, and correct me if I'm wrong, you're saying that current working **Strider** version can do the job if we place some pattern to detect a sort of tags? Like `refs/tags/[0-9]{1,3}.[0-9]{1,3}.[0-9]{1,3}-production`?

**Comment by knownasilya** [comment]
It will need some tweaking to match up the tag with the build (the regexp, and look at the right data), since the response from github is different for tags.

**Comment by cusspvz** [comment]
Could you please assign me for this? This weekend i will review entire **Strider** and **strider-github** code and PR this feature.

**Comment by knownasilya** [comment]
@cusspvz feel free to submit a PR. We can iterate till it looks good.

**Comment by knownasilya** [comment]
@cusspvz have you had any success with this?

**Comment by cusspvz** [comment]
This is currently at my urgent TODOs since we have it as a dependency. Maybe tomorrow I will dig up on this.

**Comment by cusspvz** [comment]
I've read entire `strider-github` repo already, since there are a couple of ways to apply this, i will get into this afternoon because I need to discover differences between payloads first.

**Comment by anddimario** [comment]
Any news about this problem?

**Comment by knownasilya** [comment]
I'll have a look at it this weekend.
